### ISSUE TITLE
Network traffic collector plugin?

### ISSUE THREAD

**Issue by sahilsuneja1** [cloudviz/agentless-system-crawler#226]
## Description

Can crawler support an fprobe-like plugin?

fprobe collects, converts and caches network packets into (netflow)
flows  (src/dst ip/port protocol size etc.), and periodically dumps it
to a netflow collector using netflow protocol (templates/headers
etc.). The collector can then analyze the flow characteristics for
intrusion detection or traffic profiling.

Async kind of model for fprobe (fire fprobe over veth, it periodically
scans, sleeps, sends data to collector) vs. sync model for crawler
(every iteration take snapshot of system to send to backend)

To use fprobe via crawler, need to run a local collector which holds
onto the fprobe exported flow, and forwards it to the actual collector
at each crawler iteration. If you want to use the frame equivalence
then fprobe output from local collector needs to be massaged as json,
pushed to backend db, and the frame needs to converted back to netflow
data for colector to consume.

Running fprobe via crawler would require flow forwarding logic to be
implemented as a crawler plugin (which fires fprobes and local
collector on iteration 1) . High frequency flow data transmission
might be problematic. Seems like adding inefficiency and complexity to
normal fprobe -> collector flow. If, however, fprobe supported
redirecting flows to files this would be feasible. It does not, source
would need to be changed.

## Debugging Commands Output

**Output of `docker version`:**

```
(paste your output here)
```

**Output of `docker info`:**

```
(paste your output here)
```

**Output of `python --version`:**

```
(paste your output here)
```

**Output of `pip freeze`:**

```
(paste your output here)
```

**Comment by sahilsuneja1** [comment]
Basically, two options to explore:
1. a local collector that forwards netflows to the actual external collector
2. changing fprobe to write data to ramfs so crawler can pick it up and send it to the external collector.

**Comment by sahilsuneja1** [comment]
option 1 implemented in PR #256
